### ISSUE TITLE
Support presets

### ISSUE THREAD

**Issue by praburangki** [windicss/windicss-intellisense#51]
We have a config that is used in multiple projects and we want to use the presets functionality.

Here is an example of the config, I declare it in one file for testing:

```js
const presetExample = {
  theme: {
    fontFamily: {
      sans:
        'Inter,-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Open Sans","Helvetica Neue",sans-serif',
    },
    colors: COLORS,
    borderRadius: BORDER_RADIUS,
    boxShadow: SHADOW,
  },
  plugins: [
    plugin(function ({ addUtilities }) {
      addUtilities(TYPOGRAPHY);
      addUtilities(BACKGROUND);
    }),
  ],
};

export default defineConfig({
  presets: [presetExample]
});
```

Then in vscode it shows an error:
![image](https://user-images.githubusercontent.com/8610935/111405662-d69dfc00-870b-11eb-83c3-e519420c97f9.png)

and the error log:
![image](https://user-images.githubusercontent.com/8610935/111405890-38f6fc80-870c-11eb-903b-0725f8439fea.png)

**Comment by alexanderniebuhr** [comment]
I think it is caused by `defineConfig`. I have seen this error in some of the issues here already.. cc @voorjaar

**Comment by antfu** [comment]
Where do you import `defineConfig` from?

**Comment by praburangki** [comment]
> Where do you import `defineConfig` from?

From here:
`import { defineConfig } from 'vite-plugin-windicss';`

**Comment by antfu** [comment]
@alexanderniebuhr looked at the stack trace, it does not seem be the reason of using `defineConfig`

**Comment by alexanderniebuhr** [comment]
ok i ll check later then

**Comment by alexanderniebuhr** [comment]
@antfu actually i have to understand defineConfig. It seems that it get not parsed.. so the extension sees the following on init.. so if calling `config.theme`, it gets the error this could not work, as it is expectin a config such as the example below.. i just might not get the work of defineConfig then

![config](https://user-images.githubusercontent.com/45965090/111428262-8cab1b00-86f7-11eb-8cc4-cf1492f04730.png)

```
{
  darkMode: 'class', // or 'media'
  theme: {
    screens: {
      'sm': '640px',
      'md': '768px',
      'lg': '1024px',
      'xl': '1280px',
      '2xl': '1536px',
    },
    colors: {
      gray: colors.coolGray,
      blue: colors.lightBlue,
      red: colors.rose,
      pink: colors.fuchsia,
    },
    fontFamily: {
      sans: ['Graphik', 'sans-serif'],
      serif: ['Merriweather', 'serif'],
    },
    extend: {
      spacing: {
        128: '32rem',
        144: '36rem',
      },
      borderRadius: {
        '4xl': '2rem',
      },
    },
  }
}
```

**Comment by praburangki** [comment]
> ok thanks.. i will check.. @praburangki i guess you are using typescript config??

yes correct, i've just tried converting to js using `require` & `module.exports` and got the same error.

**Comment by alexanderniebuhr** [comment]
> It seems that it get not parsed

this is the issue, have to wait for @antfu let me know how we can call `defineConfig` convert

**Comment by antfu** [comment]
@alexanderniebuhr `defineConfig` is just a bypass function for type hints, it does nothing to the config.

```ts
function defineConfig(config) {
	return config
}
```

I am not familiar with presets, but I think maybe you will need to have it passed to the Processer and expand the `presets` for you to cosume.

**Comment by alexanderniebuhr** [comment]
actually do we already support https://tailwindcss.com/docs/presets in windi css?
I want to check if that is extension exclusive bug, or also happening without extension

**Comment by hannoeru** [comment]
@alexanderniebuhr Yes, we do support presets in processor.

**Comment by alexanderniebuhr** [comment]
@hannoeru @voorjaar we use the processor, so it should convert presets right? 
https://github.com/windicss/windicss-intellisense/blob/9467683ed172d842dce0246d32e955f8148e9c7b/src/lib/core.ts#L3
https://github.com/windicss/windicss-intellisense/blob/9467683ed172d842dce0246d32e955f8148e9c7b/src/lib/core.ts#L28-L32

**Comment by hannoeru** [comment]
@alexanderniebuhr Yes, it should.

**Comment by alexanderniebuhr** [comment]
yeah will be fixed in future releaes sometime

**Comment by voorjaar** [comment]
I fixed this bug in the core two days ago, but the intellisense dependency version was not updated, probably because of this.

**Comment by voorjaar** [comment]
should be fixed in v0.6.2
